examples/src/bin/video_player.rs

```diff
@@ -52,20 +52,32 @@ mod imp {
 
         fn class_init(klass: &mut Self::Class) {
             Self::bind_template(klass);
+            klass.install_action(
+                "win.open",
+                None,
+                move |win, _action_name, _action_target| {
+                    let self_ = imp::VideoPlayerWindow::from_instance(&win);
+                    self_.dialog.set_transient_for(Some(win));
+                    self_
+                        .dialog
+                        .connect_response(clone!(@weak win => move |d, response| {
+                            if response == gtk::ResponseType::Accept {
+                                win.set_video(d.get_file().unwrap());
+                            }
+                            d.destroy();
```

Comment:
```suggestion
                            d.close();
```

Maybe? `destroy()` seems so destructive

Reply:
There's no close method for Native dialogs